core/templates/pages/contributor-dashboard-page/contributions-and-review/contributions-and-review.component.spec.ts

```diff
@@ -217,6 +259,57 @@ describe('Contributions and review component', function() {
       expect(ctrl.userIsLoggedIn).toBe(true);
       expect(ctrl.userDetailsLoading).toBe(false);
       expect(ctrl.reviewTabs.length).toEqual(2);
+      expect(ctrl.activeExplorationId).toBeNull();
+    });
+
+    it('should clear activeExplorationId when active topic changes',
+      fakeAsync(function() {
+        ctrl.onClickReviewableTranslations('explorationId');
+        expect(ctrl.activeExplorationId).toBe('explorationId');
+
+        mockActiveTopicEventEmitter.emit();
+        tick();
+
+        expect(ctrl.activeExplorationId).toBeNull();
+      }));
+
+    describe('ctrl.isReviewTranslationsTab', () => {
+      it('should return true on Review Translations tab', function() {
+        ctrl.switchToTab(ctrl.TAB_TYPE_REVIEWS, 'translate_content');
+        expect(ctrl.isReviewTranslationsTab()).toBeTrue();
```

Comment:
Here and below should there be prechecks?

Reply:
e.g. check that before switching tabs, the tab is not "review translations"

Reply:
Mmm well we're not testing the tab switching functionality here, just the output of the `isReviewTranslationsTab()` method.